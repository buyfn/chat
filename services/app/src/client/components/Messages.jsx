import React from 'react';

class Messages extends React.Component {
  componentDidUpdate() {
    window.scrollTo(0, document.body.scrollHeight);
  }

  render() {
    return (
      <div>
        <ul className="messageList">
          {this.props.messages.map(msg =>
            <li key={msg._id}><span>{msg.username}: </span>{msg.text}</li>)}
        </ul>
      </div>
    );
  }
}

export default Messages;
