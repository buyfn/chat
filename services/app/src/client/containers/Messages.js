import { connect } from 'react-redux';
import Messages from '../components/Messages.jsx';

const mapStateToProps = (state) => {
  const props = {
    messages: state.messages,
  };
  return props;
};

export default connect(mapStateToProps)(Messages);
