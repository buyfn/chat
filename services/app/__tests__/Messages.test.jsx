import React from 'react';
import renderer from 'react-test-renderer';
import Messages from '../src/client/components/Messages.jsx';

it('renders messages correctly', () => {
  const messages = [
    { username: 'anon', text: 'hi', _id: 1 },
    { username: 'vovan', text: 'zdarova', _id: 2 },
  ];
  const tree = renderer
    .create(<Messages messages={messages} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
