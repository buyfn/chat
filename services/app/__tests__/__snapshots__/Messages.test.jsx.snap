// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders messages correctly 1`] = `
<div>
  <ul
    className="messageList"
  >
    <li>
      <span>
        anon
        : 
      </span>
      hi
    </li>
    <li>
      <span>
        vovan
        : 
      </span>
      zdarova
    </li>
  </ul>
</div>
`;
